provider/core/src/main/java/org/arquillian/pact/provider/core/InteractionRunner.java

```diff
@@ -125,6 +128,25 @@ private void setField(Object testInstance, Field fieldTarget, Object pact) {
         }
     }
 
+    protected void validateState(final TestClass testClass, final List<Throwable> errors) {
+        Arrays.stream(testClass.getMethods(State.class)).forEach(method -> {
+            validatePublicVoidMethods(method, errors);
+            if (isMethodWithSingleMapParameter(method)) {
+                final String mapError = String.format("Method %s should take only a single Map parameter", method.getName());
+                logger.log(Level.SEVERE, mapError);
+                errors.add(new IllegalArgumentException(mapError));
```

Comment:
Not sure if I understand what is going on here. `isMethodWithSingleMapParameter` should be probably `methodDoesNotHaveSingleMapParameter`?